### DISCUSSION TITLE
What is the point of calling the same method 2 times with the same parameters?

### DISCUSSION THREAD

**Discussion by korenevskiy** [joomla/joomla-cms#36425] [Q&A] [answered]
https://github.com/joomla/joomla-cms/blob/63d4e65cda6044e02f4b06adae104f56db882a09/administrator/components/com_fields/src/Plugin/FieldsPlugin.php#L149
https://github.com/joomla/joomla-cms/blob/63d4e65cda6044e02f4b06adae104f56db882a09/administrator/components/com_fields/src/Plugin/FieldsPlugin.php#L153

**Comment by PhilETaylor** [comment]
Nothing to do with me. Talk to @laoneo

References: 

 - https://github.com/joomla/joomla-cms/pull/28516#issuecomment-607263719

 - https://github.com/joomla/joomla-cms/pull/28534 
 - https://github.com/joomla/joomla-cms/pull/28516

**Reply by korenevskiy** [reply]
Thanks. It's just that you are listed there as a contributor.

**Comment by korenevskiy** [comment]
@astridx @laoneo  @astridx

**Comment by astridx** [comment]
@korenevskiy Did you see the conversation in this PR https://github.com/joomla/joomla-cms/pull/28516 ?

**Comment by laoneo** [comment]
I guess this is a bug as it was wrongly merged up from 3. Compare the two files:
- https://github.com/joomla/joomla-cms/blob/4.0-dev/administrator/components/com_fields/src/Plugin/FieldsPlugin.php#L149
- https://github.com/joomla/joomla-cms/blob/3.10-dev/administrator/components/com_fields/libraries/fieldsplugin.php#L125
When you look on the original pr #28534, then you see that the whole of block is not there. It was only added in 3 for BC reasons.

**Comment by korenevskiy** [comment]
@laoneo @astridx 
In the final release of Joomla (**Joomla! 4.0.5 Stable** 11-December-2021), I wrote a component using plugins. and I came across this code.
I think it's a mistake, but I didn't figure it out. You will mean this.

**Comment by laoneo** [comment, marked as answer]
Made a pr which removes the duplicated code in #36434. You can mark this question as answered.
